pennylane/tape/tape.py

```diff
@@ -872,6 +872,145 @@ def set_parameters(self, params, trainable_only=True):
             op = self._par_info[idx]["op"]
             op.data[self._par_info[idx]["p_idx"]] = p
 
+    @staticmethod
+    def _single_measurement_shape(measurement_process, device):
+        """
+
+        Note: these shapes and dtypes are specific to tapes and may differ from
+        QNode outputs.
+        """
+
+        shape = tuple()
+
+        output_domain = "real"
+        ret_type = measurement_process.return_type
+        if ret_type in (qml.operation.Expectation, qml.operation.Variance):
+
+            shape = (1,)
+
+        elif ret_type == qml.operation.Probability:
+
+            dim = 2 ** len(measurement_process.wires)
+            shape = (1, dim)
```

Comment:
note that for a CV device, this will be `cutoff ** len(measurement_process.wires)`!

Reply:
Thanks! I've added consideration for this and a test.